### PR TITLE
2284 support page translations

### PR DESCRIPTION
**Pull request by paulmsmith**
## Proposed changes

[Ticket here](https://trello.com/c/yigrEoJy/2284-abstract-content-from-support-pages-in-funding-frontend)

### CODE REVIEW

File: app/views/support/report_a_problem.html.erb [hunk 1]
```diff
@@ -27,14 +27,14 @@
...
-          Report a problem
+          <%= t('report_a_problem.headings.1') %>
         </h1>
 
       </legend>
 
       <div class="govuk-inset-text">
-        <a href="https://www.heritagefund.org.uk/service-status">Check our service status page</a>
-        to see if there are any known issues.
+        <a href="https://www.heritagefund.org.uk/service-status"><%= t('report_a_problem.text.service_status.link_text') %></a>
```
Comment (stuartmccoll) [inline app/views/support/report_a_problem.html.erb:36]: There's a better pattern for sentences which contain links:

The `.yml` could look like the following:

```yml
service_status_html: "%{href} to see if there are any known issues."
service_status_text: "Check our service status page"
```

Then you can refer to this in the view in the following way:

```ruby
<%=
  t(
    "report_a_problem.text.service_status_html",
    href: link_to(
            t("report_a_problem.text.service_status_text"),
            "https://www.heritagefund.org.uk/service-status",
            target: "_blank",
            class: "govuk-link"
    )
  )
%>
```

File: app/views/support/question_or_feedback.html.erb [hunk 2]
```diff
@@ -52,12 +52,12 @@
...
 
-      <h2 class="govuk-heading-m">Do you want a reply?</h2>
+      <h2 class="govuk-heading-m"><%= t('question_or_feedback.headings.do_you_want_a_reply') %></h2>
 
-      <p class="govuk-body">Leave your details below if you’d like a response.</p>
+      <p class="govuk-body"><%= t('question_or_feedback.text.leave_your_details_below') %></p>
 
       <div class="govuk-form-group">
-        <%= label_tag :support_question_or_feedback_name, "Your name", class: "govuk-label" %>
+        <%= label_tag :support_question_or_feedback_name, t('question_or_feedback.text.your_name'), class: "govuk-label" %>
```
Comment (stuartmccoll) [inline app/views/support/question_or_feedback.html.erb:60]: We're pulling the label translations out into a parent key of `text` in this view, but pulling them out into a parent key of `form` in the 'report a problem' view. Could we go for a parent key of `labels` for both

File: app/views/support/question_or_feedback.html.erb [hunk 3]
```diff
@@ -81,10 +81,9 @@
   </div>
 
   <%=
-    submit_tag "Send",
+    submit_tag t('question_or_feedback.buttons.send'),
```
Comment (stuartmccoll) [inline app/views/support/question_or_feedback.html.erb:84]: A good candidate for the `generic` section in `en.yml` 🙂.

File: app/views/support/question_or_feedback.html.erb [hunk 4]
```diff
@@ -27,7 +27,7 @@
       <legend class="govuk-fieldset__legend govuk-fieldset__legend--xl">
 
         <h1 class="govuk-fieldset__heading">
-          Ask a question or give feedback
+          <%= t('question_or_feedback.headings.ask_a_question_or_give_feedback') %>
```
Comment (stuartmccoll) [inline app/views/support/question_or_feedback.html.erb:30]: We've tried to be consistent in using `page_heading` as the locale key for the `h1` element across most of the service.

File: app/views/support/question_or_feedback.html.erb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <%
   content_for :page_title,
-              flash[:errors].present? ? "Error: Ask a question or give feedback" : "Ask a question or give feedback"
+              flash[:errors].present? ? (t('generic.error') + ": " + t('question_or_feedback.text.ask_a_question_or_give_feedback')).html_safe : (t('question_or_feedback.text.ask_a_question_or_give_feedback')).html_safe
```
Comment (stuartmccoll) [inline app/views/support/question_or_feedback.html.erb:3]: We've been using `page_title` as the locale key for the `title` element across most of the service. Using something different from the `h1` element's locale key allows us to change them easily if we decide that we want the title to say something different, for example.

File: app/views/support/show.html.erb [hunk 6]
```diff
@@ -51,38 +51,50 @@
   </div>
 
   <%=
-    submit_tag "Continue",
+    submit_tag t('support.buttons.continue'),
```
Comment (stuartmccoll) [inline app/views/support/show.html.erb:54]: Another good candidate for abstracting into the `generic` section of the app's main `en.yml`.

**Review by stuartmccoll — CHANGES_REQUESTED** [review]

File: app/views/support/show.html.erb [hunk 7]
```diff
@@ -51,38 +51,50 @@
...
+      <li>
+        <a href="https://www.heritagefund.org.uk/funding/outcomes">
+          <%= t('support.links.2') %>
+        </a>
+      </li>
+      <li>
+        <a href="https://www.heritagefund.org.uk/funding">
+          <%= t('support.links.3') %>
+        </a>
+      </li>
```
Comment (ptrelease) [inline app/views/support/show.html.erb:90]: Did we agree to use the rails link to here?

Comment (ptrelease) [reply app/views/support/show.html.erb:90]: Will pick up in a separate piece of work at an appropriate time.

**Review by ptrelease — APPROVED** [review]

**Pull request merged**